newIDE/app/src/EventsSheet/index.js

```diff
@@ -1105,85 +1126,106 @@ export default class EventsSheet extends React.Component<Props, State> {
+                                enabled: this._selectionCanHaveSubEvents(),
+                              },
+                              {
+                                label: 'Add Other',
+                                submenu: this.state.allEventsMetadata.map(
+                                  metadata => {
+                                    return {
+                                      label: metadata.fullName,
+                                      click: () =>
+                                        this.addNewEvent(metadata.type),
+                                    };
+                                  }
                                 ),
-                            },
-                            {
-                              label: 'Add Sub Event',
-                              click: () => this.addSubEvents(),
-                              enabled: this._selectionCanHaveSubEvents(),
-                            },
-                            {
-                              label: 'Add Other',
-                              submenu: this.state.allEventsMetadata.map(
-                                metadata => {
-                                  return {
-                                    label: metadata.fullName,
-                                    click: () =>
-                                      this.addNewEvent(metadata.type),
-                                  };
+                              },
+                              { type: 'separator' },
+                              {
+                                label: 'Undo',
+                                click: this.undo,
+                                enabled: canUndo(this.state.history),
+                                accelerator: 'CmdOrCtrl+Z',
+                              },
+                              {
+                                label: 'Redo',
+                                click: this.redo,
+                                enabled: canRedo(this.state.history),
+                                accelerator: 'CmdOrCtrl+Shift+Z',
+                              },
+                              { type: 'separator' },
+                              {
+                                label: 'Extract Events to a Function',
+                                click: () => this.extractEventsToFunction(),
+                              },
+                              {
+                                label: 'Move Events into a Group',
+                                click: () => this.moveEventsIntoNewGroup(),
+                              },
+                              {
+                                label: 'Analyze Objects Used in this Event',
+                                click: this._openEventsContextAnalyzer,
+                              },
+                            ];
+
+                            const edition = {
+                              label: 'Edit',
```

Comment:
Do you think we should be more precise "Edit Text and Color(s)"?

Reply:
No, the sub-menus must be concise.